app/src/lib/git/branch.ts

```diff
@@ -64,9 +64,64 @@ export async function deleteLocalBranch(
   return true
 }
 
+export async function deleteRemoteBranch(
+  repository: Repository,
+  branch: Branch,
+  account: IGitAccount | null
+): Promise<true> {
+  const remoteName =
+    branch.type === BranchType.Remote
+      ? branch.remoteName
+      : branch.upstreamRemoteName
+
+  // This should not happen - a remote branch should have a remote.
+  if (remoteName === null) {
+    log.error(`Could not determine the branch's remote ${remoteName}`)
+    return true
+  }
+
+  const networkArguments = await gitNetworkArguments(repository, account)
+  const remoteUrl =
+    (await getRemoteURL(repository, remoteName).catch(err => {
+      // If we can't get the URL then it's very unlikely Git will be able to
+      // either and the push will fail. The URL is only used to resolve the
+      // proxy though so it's not critical.
+      log.error(`Could not resolve remote url for remote ${remoteName}`, err)
+      return null
+    })) || getFallbackUrlForProxyResolve(account, repository)
+
+  const args = [
+    ...networkArguments,
+    'push',
+    remoteName,
+    `:${branch.nameWithoutRemote}`,
```

Comment:
Nice! 

Since this is a change to the original functionality, I will open a PR just to address it (probably as an on top of branch.ts refactor branch) and test it out including the scenario about mistakenly deleting the wrong remote branch.

Reply:
PR https://github.com/desktop/desktop/pull/11483 now uses upstreamWithoutRemote when deleting remotes.